In [33]:
# importing libraries
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [34]:
# loading the datasets
df_trips_company = pd.read_csv('data/trips_by_company.csv')
df_trips_neighborhood = pd.read_csv(
    'data/avg_trips_by_dropoff_neighborhood.csv')

In [35]:
df_trips_company.info()

<class 'pandas.DataFrame'>
RangeIndex: 64 entries, 0 to 63
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   company_name  64 non-null     str  
 1   trips_amount  64 non-null     int64
dtypes: int64(1), str(1)
memory usage: 1.1 KB


- dataframe with 2 columns and 64 rows;
- no null values;
- data types are accurate.

In [36]:
df_trips_company.describe()

,trips_amount
count,64.000000
mean,2145.484375
std,3812.310186
min,2.000000
25%,20.750000
50%,178.500000
75%,2106.500000
max,19558.000000


- mean (2145) >> median (178) → strong right skew;
- std (3812) > mean → high dispersion driven by top companies;
- #1 trips company leads with 19,558 trips, most companies have far fewer.

In [37]:
print(df_trips_company.head(10))

                        company_name  trips_amount
0                          Flash Cab         19558
1          Taxi Affiliation Services         11422
2                   Medallion Leasin         10367
3                         Yellow Cab          9888
4    Taxi Affiliation Service Yellow          9299
5          Chicago Carriage Cab Corp          9181
6                       City Service          8448
7                           Sun Taxi          7701
8          Star North Management LLC          7455
9  Blue Ribbon Taxi Association Inc.          5953


- Data is sorted by trips_amount so these are the top 10 companies;
- top 10 companies account for the bulk of all trips, showing high market concentration.

In [38]:
df_trips_neighborhood.info()

<class 'pandas.DataFrame'>
RangeIndex: 94 entries, 0 to 93
Data columns (total 2 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   dropoff_location_name  94 non-null     str    
 1   average_trips          94 non-null     float64
dtypes: float64(1), str(1)
memory usage: 1.6 KB


- dataframe with 2 columns and 94 rows;
- no null values;
- data types are accurate.

In [39]:
df_trips_neighborhood.describe()

,average_trips
count,94.000000
mean,599.953728
std,1714.591098
min,1.800000
25%,14.266667
50%,52.016667
75%,298.858333
max,10727.466667


- mean (599.95) >> median (52.02) → strong right skew;
- std (1714.59) > mean → high dispersion driven by top destination neighborhoods;
- #1 dropoff neighborhood leads with 10,727.46 avg trips, most neighborhoods have far fewer.

In [40]:
print(df_trips_neighborhood.head(10))

  dropoff_location_name  average_trips
0                  Loop   10727.466667
1           River North    9523.666667
2         Streeterville    6664.666667
3             West Loop    5163.666667
4                O'Hare    2546.900000
5             Lake View    2420.966667
6            Grant Park    2068.533333
7         Museum Campus    1510.000000
8            Gold Coast    1364.233333
9    Sheffield & DePaul    1259.766667


- Data is sorted by average_trips so these are the top 10 dropoff locations;
- top 6 dropoff locations have significantly higher average trips than the rest.

In [ ]:
top10_companies = df_trips_company.head(10)

fig, ax = plt.subplots(figsize=(10, 6))
sns.barplot(data=top10_companies, x='trips_amount', y='company_name', ax=ax)
ax.bar_label(ax.containers[0], fmt='%.0f', padding=3)
ax.set_title('Top 10 Taxi Companies by Number of Rides')
ax.set_xlabel('Number of Rides')
ax.set_ylabel('Company')
plt.tight_layout()
plt.show()

Flash Cab clearly leads the market with 19,558 rides — approximately 70% more than the second-ranked company, Taxi Affiliation Services. The remaining nine companies are clustered within a narrower range of 6,000 to 11,000 rides, with no dramatic differences between them. This distribution suggests a market with a standout leader and a group of similarly sized competitors. It would be worth investigating whether Flash Cab's dominance remains consistent over time or whether it is driven by specific periods.

In [ ]:
top10_neighborhoods = df_trips_neighborhood.head(10)

fig, ax = plt.subplots(figsize=(10, 6))
sns.barplot(data=top10_neighborhoods, x='average_trips', y='dropoff_location_name', ax=ax)
ax.bar_label(ax.containers[0], fmt='%.0f', padding=3)
ax.set_title('Top 10 Dropoff Neighborhoods by Average Number of Rides')
ax.set_xlabel('Average Number of Rides')
ax.set_ylabel('Neighborhood')
plt.tight_layout()
plt.show()

Loop and River North stand out as the top dropoff neighborhoods, averaging approximately 11,000 and 9,500 rides per observation period respectively — nearly double the third-ranked neighborhood, Streeterville. The data naturally segments into three groups: Loop and River North as clear leaders, a mid-tier cluster including Streeterville and West Loop ranging between 5,000 and 7,000 average rides, and a lower-demand group of six neighborhoods averaging fewer than 2,500 rides. The dominance of Loop and River North is consistent with their urban profile — Loop is Chicago's central business district, generating high volumes of weekday commuter and business trips, while River North's restaurant and nightlife scene drives evening demand. Notably, O'Hare Airport's presence in the top 10 suggests a distinct travel pattern driven by business travellers and tourists rather than local leisure trips.